### ISSUE TITLE
[Bug]: Bad sad error in SemanticAnalyzer

### ISSUE THREAD

**Issue by mindula** [ballerina-platform/ballerina-lang#41101]
### Description

Consider the following scenario,

https://github.com/ballerina-platform/ballerina-lang/assets/61020198/2401e0b6-7647-4d62-a005-5d5e7a8a3eb6

This can also be seen when doing a bal build,

<img width="946" alt="Screenshot 2023-07-27 at 12 13 32" src="https://github.com/ballerina-platform/ballerina-lang/assets/61020198/7efb2410-0a15-42ba-b702-b025dae14955">

### Steps to Reproduce

```
var orders = [{buyer: "b1", price: 10}];

var buyers = from var {buyer, price} in orders group by var item = 
```

### Related area

-> Compilation

**Comment by KavinduZoysa** [comment]
We can fix this issue by adding following check to [this line](https://github.com/ballerina-platform/ballerina-lang/blob/ce8bce4ae63c4e6ef97909b1fe53af74b9528591/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java#L1876).
```java
(ownerSymTag & SymTag.PACKAGE) == SymTag.PACKAGE
```
But we need to check whether this is the correct approach or not.

**Comment by mindula** [comment]
This is fixed in 2201.9.2. Hence, closing the issue.
